src/main/java/io/camunda/testing/assertions/ProcessInstanceAssertions.java

```diff
@@ -0,0 +1,413 @@
+            .findFirst();
+
+    assertThat(processInstanceRecord.isPresent())
+        .withFailMessage("Process with key %s was not completed", actual.getProcessInstanceKey())
+        .isTrue();
+
+    return this;
+  }
+
+  /**
+   * Verifies the expectation that the process instance is not completed.
+   *
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions isNotCompleted() {
+    final Optional<Record<ProcessInstanceRecordValue>> processInstanceRecord =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey())
+            .withBpmnElementType(BpmnElementType.PROCESS)
+            .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED).stream()
+            .findFirst();
+
+    assertThat(processInstanceRecord.isPresent())
+        .withFailMessage("Process with key %s was completed", actual.getProcessInstanceKey())
+        .isFalse();
+
+    return this;
+  }
+
+  /**
+   * Verifies the expectation that the process instance is terminated.
+   *
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions isTerminated() {
+    final Optional<Record<ProcessInstanceRecordValue>> processInstanceRecord =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey())
+            .withBpmnElementType(BpmnElementType.PROCESS)
+            .withIntent(ProcessInstanceIntent.ELEMENT_TERMINATED).stream()
+            .findFirst();
+
+    assertThat(processInstanceRecord.isPresent())
+        .withFailMessage("Process with key %s was not terminated", actual.getProcessInstanceKey())
+        .isTrue();
+
+    return this;
+  }
+
+  /**
+   * Verifies the expectation that the process instance is not terminated.
+   *
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions isNotTerminated() {
+    final Optional<Record<ProcessInstanceRecordValue>> processInstanceRecord =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey())
+            .withBpmnElementType(BpmnElementType.PROCESS)
+            .withIntent(ProcessInstanceIntent.ELEMENT_TERMINATED).stream()
+            .findFirst();
+
+    assertThat(processInstanceRecord.isPresent())
+        .withFailMessage("Process with key %s was terminated", actual.getProcessInstanceKey())
+        .isFalse();
+
+    return this;
+  }
+
+  /**
+   * Verifies the expectation that the process instance has passed an element with a specific
+   * element id exactly one time.
+   *
+   * @param elementId The id of the element
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions hasPassedElement(final String elementId) {
```

Comment:
:thought_balloon: Maybe rename to `hasPassedExactlyOnce()`

Reply:
I prefer this name and feel like mentioning it in the javadoc would be sufficient. The reason I've added `hasPassedElement` and `hasNotPassedElement` is because it makes the unit test more readable in my opinion. If we choose to rename it I'd rather remove it entirely and encourage people to use `hasPassedElement(elementId, times)`.